crates/solidity/testing/snapshots/cst_output/Block/unchecked/generated/0.8.0-success.yml

```diff
@@ -6,25 +6,26 @@ Source: >
 Errors: []
 
 Tree:
-  - Block: # 0..24 "{ unchecked { x = 1; } }"
-      - (open_brace) OpenBrace: "{" # 0..1
-      - (statements) Statements: # 1..22 " unchecked { x = 1; }"
-          - (item) Statement: # 1..22 " unchecked { x = 1; }"
-              - (variant) UncheckedBlock: # 1..22 " unchecked { x = 1; }"
-                  - (unchecked_keyword) UncheckedKeyword: "unchecked" # 2..11
-                  - (block) Block: # 11..22 " { x = 1; }"
-                      - (open_brace) OpenBrace: "{" # 12..13
-                      - (statements) Statements: # 13..20 " x = 1;"
-                          - (item) Statement: # 13..20 " x = 1;"
-                              - (variant) ExpressionStatement: # 13..20 " x = 1;"
-                                  - (expression) Expression: # 13..19 " x = 1"
-                                      - (variant) AssignmentExpression: # 13..19 " x = 1"
-                                          - (left_operand) Expression: # 13..15 " x"
-                                              - (variant) Identifier: "x" # 14..15
-                                          - (operator) Equal: "=" # 16..17
-                                          - (right_operand) Expression: # 17..19 " 1"
-                                              - (variant) DecimalNumberExpression: # 17..19 " 1"
-                                                  - (literal) DecimalLiteral: "1" # 18..19
-                                  - (semicolon) Semicolon: ";" # 19..20
-                      - (close_brace) CloseBrace: "}" # 21..22
-      - (close_brace) CloseBrace: "}" # 23..24
+  - (Block): # 0..24 "{ unchecked { x = 1; } }"
+      - (open_brace꞉ OpenBrace): "{" # 0..1
+      - (statements꞉ Statements) ► (item꞉ Statement) ► (variant꞉ UncheckedBlock): # 1..22 " unchecked { x = 1; }"
```

Comment:
The flattened path immediately makes sense to me for some terminal path
```
- (LeadingTrivia) ► (Whitespace): " " # 5..6
```

but when used in a non-terminal path/a node that has children it's confusing for me, for whatever reason. Maybe it's because it combines quite a bit of glyphs/sigils but also contains rendered children content that may be mistook visually for YAML string value? i.e. `# 1..22 " unchecked { x = 1; }"` Maybe using different characters like 《content》to disambiguate would help?

I also tried to experiment with piggy-backing on the file structure shorthand forms (used in GitHub interface, VS Code): `some/node` but the name metadata makes this significantly harder to read.

I don't have more alternatives and I know this is subjective...

Alternatively, maybe we could at first only enable the flattened/shorthand path for terminal paths/nodes and see if it's good enough for now?
This would keep the tree layout consistent, i.e. disallow following confusing structure:
```
- parent
    - child > grandchild:
        - great_grandchild: "value"
    - child:
        - grandchild: "value"
```

Or maybe doing this not only for terminals as per above but also for always-single-child-nodes (enums) but not for lists/other with 1 child would strike a better balance here?

For example, this might be too big of a shortcut that IMO hurts rather than improves readability:
```
- (items꞉ TupleValues) ► (item꞉ TupleValue) ► (expression꞉ Expression) ► (variant꞉ EqualityExpression): # 1..11 "foo == bar"
```

Reply:
> Maybe using different characters like 《content》to disambiguate would help?

I like those alternative delimiters. But then I did spend years writing APL, so ... :)